code/ui/blocks/src/blocks/Controls.tsx

```diff
@@ -59,16 +59,19 @@ export const Controls: FC<ControlsProps> = (props) => {
   const hasSubcomponents = Boolean(subcomponents) && Object.keys(subcomponents).length > 0;
 
   if (!hasSubcomponents) {
-    return (
-      <PureArgsTable
-        rows={filteredArgTypes}
-        sort={sort}
-        args={args}
-        globals={globals}
-        updateArgs={updateArgs}
-        resetArgs={resetArgs}
-      />
-    );
+    if (Object.keys(filteredArgTypes).length > 0 || Object.keys(args).length > 0) {
```

Comment:
I think we need both, as even without argTypes we will show controls based on arg values.